ykcompile/src/lib.rs

```diff
@@ -118,16 +123,29 @@ pub struct TraceCompiler {
     /// Stores the destination local of the outermost function and moves its content into RAX at
     /// the end of the trace.
     rtn_var: Option<Place>,
+    /// Local referencing the input arguments to the trace.
+    trace_inputs_local: Option<Local>,
     /// Stack builder for allocating objects on the stack.
     stack_builder: StackBuilder,
+    _pd: PhantomData<TT>,
 }
 
-impl TraceCompiler {
+impl<TT> TraceCompiler<TT> {
+
     fn place_to_location(&mut self, p: &Place) -> Result<Location, CompileError> {
         if !p.projection.is_empty() {
-            return Err(CompileError::Unimplemented(format!("{}", p)));
+            if Some(p.local) == self.trace_inputs_local {
+                match &p.projection[0] {
+                    // This is a hack since we have not implement proper projection lowering yet.
```

Comment:
Perhaps make that a FIXME?

Reply:
Whoops, that's an old comment. I just needs to go.